version: '3.11'

services:
  redis:
    image: redis
    ports:
      - "6379:6379"
  db:
    restart: always
    image: postgres:16-alpine
    env_file:
      - .env
    volumes:
      - ./postgres_data:/var/lib/postgresql/data
    ports:
      - '5432:5432'
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}" ]
      interval: 5s
      timeout: 5s
      retries: 10

  app:
    build: .
    tty: true
    command: bash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    ports:
      - "8000:8000"
    volumes:
      - .:/code
    depends_on:
      db:
        condition: service_healthy

  celery:
    build: .
    tty: true
    command: celery -A config worker -l info -P eventlet
    depends_on:
      - redis
      - app

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l info
    depends_on:
      - redis
      - app
      - celery

volumes:
  pg_data: